### PR TITLE
boards: shields: Add Ambiq Apollo4p Display Shield

### PR DESCRIPTION
**Pull request by msierszulski**
This PR adds support for apollo4p_evb_disp_shield.

### CODE REVIEW

**Review by gmarull — CHANGES_REQUESTED** [review]

File: boards/shields/apollo4p_evb_disp_shield/Kconfig.defconfig [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2023 Antmicro <www.antmicro.com>
+# SPDX-License-Identifier: Apache-2.0
+
+if SHIELD_APOLLO4P_EVB_DISP_SHIELD
+
+config SPI
+	default y
+
+endif # SHIELD_APOLLO4P_EVB_DISP_SHIELD
```
[the remark this answers is not in the document]
Comment (msierszulski) [reply boards/shields/apollo4p_evb_disp_shield/Kconfig.defconfig]: Good catch, thanks!

File: boards/shields/apollo4p_evb_disp_shield/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+Programming
+***********
+
+Set ``-DSHIELD=apollo4p_evb_disp_shield`` when you invoke ``west build``. For example:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/sensor/accel_polling
+   :board: apollo4p_evb
+   :shield: apollo4p_evb_disp_shield
+   :goals: build
```
[the remark this answers is not in the document]
Comment (msierszulski) [reply boards/shields/apollo4p_evb_disp_shield/doc/index.rst]: Removed 
```Set ``-DSHIELD=apollo4p_evb_disp_shield`` when you invoke ``west build``. For example:```

File: boards/shields/apollo4p_evb_disp_shield/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+This shield can only be used with a :ref:`apollo4p_evb` board.
+
+Programming
+***********
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/sensor/accel_polling
+   :board: apollo4p_evb
+   :shield: apollo4p_evb_disp_shield
+   :goals: build
```
Comment (gmarull) [inline boards/shields/apollo4p_evb_disp_shield/doc/index.rst:48]: still redundant

Comment (msierszulski) [reply boards/shields/apollo4p_evb_disp_shield/doc/index.rst:48]: I don't get it, can you explain it?
You mean I can opt out of this whole `Programming` section? All other shields seem to have it

Comment (gmarull) [reply boards/shields/apollo4p_evb_disp_shield/doc/index.rst:48]: There's no point in documenting something that is generic in shield docs, ie, how to program. Also, referencing a specific sample is misleading (`samples/sensor/accel_polling`). Other shields should also drop this.

Comment (msierszulski) [reply boards/shields/apollo4p_evb_disp_shield/doc/index.rst:48]: Got it. Thanks

**Review by tgorochowik — APPROVED** [review]

File: boards/shields/apollo4p_evb_disp_shield/apollo4p_evb_disp_shield.overlay [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	aliases {
+		accel0 = &adxl362;
+	};
+};
+
+&iom1 {
```
Comment (jfischer-no) [inline boards/shields/apollo4p_evb_disp_shield/apollo4p_evb_disp_shield.overlay:13]: https://docs.zephyrproject.org/latest/hardware/porting/shields.html#board-compatibility

https://github.com/zephyrproject-rtos/zephyr/blob/ef5d0fd291568e39d55ef95792e217146d6c2043/boards/arm/apollo4p_evb/apollo4p_evb.dts#L37-L51

No "well-known connectors" -> not acceptable PR.

**Review by jfischer-no — CHANGES_REQUESTED** [review]

**Pull request closed without merging**